### PR TITLE
vagrant: add Vagrantfile to test native in FreeBSD

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This provides a Vagrantfile for testing RIOT `native` in FreeBSD.

### Testing procedure

Basically, follow the provided README.md. For the `tapsetup` stuff to work #14457 is required.

I also ran `dist/tools/compile_and_run_for_board` on it overnight, the results can be found [here](https://github.com/miri64/RIOT_2020.07_test_results/blob/master/2020.07-RC1pre-6fc72be72/native%40freebsd/failuresummary.md).

Basically there are some errors (but this does not block this PR), that can be summarized as such:

- Cross-compiling C++ for 32-bit (this seems to be not easily fixable at least not just by installing from `ports`, maybe another argument for #6603)
- Missing names like errno-values or GNU/Linux-specific functions such as [backtrace](https://linux.die.net/man/3/backtrace)
- Missing shell commands like `sha512sum` or `bridge` and other missing tooling
- POSIX definition clashes for `pthread`s
- Timing errors (maybe related to https://github.com/RIOT-OS/RIOT/issues/10073)
- Some minor unexpected scheduling behavior [as it seems](https://github.com/miri64/RIOT_2020.07_test_results/blob/master/2020.07-RC1pre-6fc72be72/native%40freebsd/tests/cond_order/test.failed)
- Missing root privileges for tests that require it (easily fixed for non-automated runs ;-))
- Encoding formats that already in the past yielded different results for different toolchains like [cayenne-lpp](https://github.com/miri64/RIOT_2020.07_test_results/blob/master/2020.07-RC1pre-6fc72be72/native%40freebsd/tests/pkg_cayenne-lpp/test.failed)
- Incompatibilities in OS-specific `scapy` sublayer
- ...

All fixes that can be tackled as follow-ups (I will open a tracking issue if desired).

### Issues/PRs references
None

### CODE REVIEW

**Comment by miri64** [comment]
Rebased and squashed to current master.

**Comment by leandrolanzieri** [comment]
When trying to run the network test indicated in the README I get the following:
```
make: Entering directory '/home/vagrant/RIOT/tests/gnrc_dhcpv6_client_6lbr'


/home/vagrant/RIOT/tests/gnrc_dhcpv6_client_6lbr/bin/native/tests_gnrc_dhcpv6_client_6lbr.elf -z [::1]:17754 tap0 
RIOT native interrupts/signals initialized.
LED_RED_OFF
LED_GREEN_ON
RIOT native board initialized.
RIOT native hardware initialization complete.

DHCPv6: Selecting interface 7 as upstream
main(): This is RIOT! (Version: 2020.10-devel-772-gf8d25-pr/no_archives)
> 
> 
> Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.7/site-packages/scapy/sendrecv.py", line 842, in _run
    *arg, **karg)] = iface
  File "/usr/local/lib/python3.7/site-packages/scapy/arch/bpf/supersocket.py", line 239, in __init__
    super(L2bpfListenSocket, self).__init__(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/scapy/arch/bpf/supersocket.py", line 62, in __init__
    (self.ins, self.dev_bpf) = get_dev_bpf()
  File "/usr/local/lib/python3.7/site-packages/scapy/arch/bpf/core.py", line 115, in get_dev_bpf
    raise Scapy_Exception("No /dev/bpf handle is available !")
scapy.error.Scapy_Exception: No /dev/bpf handle is available !

Exception ignored in: <function _L2bpfSocket.__del__ at 0x80271d830>
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/scapy/arch/bpf/supersocket.py", line 136, in __del__
    self.close()
  File "/usr/local/lib/python3.7/site-packages/scapy/arch/bpf/supersocket.py", line 208, in close
    if not self.closed and self.ins is not None:
AttributeError: 'L2bpfListenSocket' object has no attribute 'ins'

Traceback (most recent call last):
  File "/home/vagrant/RIOT/tests/gnrc_dhcpv6_client_6lbr/tests/01-run.py", line 173, in <module>
    sys.exit(run(testfunc, timeout=TIMEOUT, echo=True))
  File "/home/vagrant/RIOT/dist/pythonlibs/testrunner/__init__.py", line 30, in run
    testfunc(child)
  File "/home/vagrant/RIOT/tests/gnrc_dhcpv6_client_6lbr/tests/01-run.py", line 114, in testfunc
    pkt = wait_for_dhcpv6_pkt(iface)
  File "/home/vagrant/RIOT/tests/gnrc_dhcpv6_client_6lbr/tests/01-run.py", line 95, in wait_for_dhcpv6_pkt
    raise TimeoutError("Sniffing for DHCPv6 traffic timed out")
TimeoutError: Sniffing for DHCPv6 traffic timed out
make: *** [/home/vagrant/RIOT/tests/gnrc_dhcpv6_client_6lbr/../../Makefile.include:746: test] Error 1
make: Leaving directory '/home/vagrant/RIOT/tests/gnrc_dhcpv6_client_6lbr'
```

**Comment by leandrolanzieri** [comment]
Please squash @miri64 !

**Review by leandrolanzieri — APPROVED** [review]
I followed the instructions and got it working, also @miri64 has provided test output. Some follow-up work is needed but I see no reason not to merge this. ACK.

File: dist/tools/vagrant/freebsd/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  if ! [ -d /home/vagrant/RIOT ]; then
+      git clone --recurse-submodules https://github.com/RIOT-OS/RIOT /home/vagrant/RIOT
+  fi
+  chown -R vagrant: /home/vagrant/RIOT
+  curl https://raw.githubusercontent.com/RIOT-OS/riotdocker/master/requirements.txt \
+      > /tmp/requirements.txt
+  su - vagrant -c "pip install --user -r /tmp/requirements.txt"
+  # install most current pexpect to prevent async bug
+  su - vagrant -c "pip install --upgrade --user pexpect"
+  grep -q "export MAKE=gmake" /home/vagrant/.profile || \
```
Comment (leandrolanzieri) [inline dist/tools/vagrant/freebsd/Vagrantfile:16]: The build system sanity check does not like these exports, this file needs to be whitelisted somehow

**Comment by miri64** [comment]
Squashed and added whitelisting to the build system sanity check.

**Pull request merged**